File: 0941. Valid Mountain Array/Solution.py
from typing import List


class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        i, j = 0, len(A) - 1
        while i < j and A[i] < A[i + 1]:
            i += 1
        while i < j and A[j] < A[j - 1]:
            j -= 1
        return 0 < i == j < len(A) - 1

File: 0941. Valid Mountain Array/Solution.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    bool validMountainArray(vector<int> &A)
    {
        int i = 0, j = A.size() - 1;
        while (i < j && A[i] < A[i + 1])
            i++;
        while (j > i && A[j] < A[j - 1])
            j--;
        return i > 0 && j < A.size() - 1 && i == j;
    }
};
